artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PersistentDuplicateIDCache.java

```diff
@@ -240,6 +242,7 @@ private synchronized void addToCache(final ByteArray holder,
          }
 
          if (instantAdd) {
+            addToCacheInMemory(holder, recordID);
```

Comment:
I don't get it.. wouldn't this be either in TX or not?

Reply:
yes but it isn't executed by TX if instantAdd = true